internal/tools/looker/lookercommon/lookercommon.go

```diff
@@ -262,3 +264,40 @@ func ProcessQueryArgs(ctx context.Context, params tools.ParamValues) (*v4.WriteQ
 	}
 	return &wq, nil
 }
+
+type QueryApiClientContext struct {
+	Name            string            `json:"name"`
+	Attributes      map[string]string `json:"attributes,omitempty"`
+	ExtraAttributes map[string]string `json:"extra_attributes,omitempty"`
+}
+
+type RenderOptions struct {
+	Format string `json:"format"`
+}
+
+type RequestRunInlineQuery2 struct {
+	Query             v4.WriteQuery         `json:"query"`
+	RenderOpts        RenderOptions         `json:"render_options"`
+	QueryApiClientCtx QueryApiClientContext `json:"query_api_client_context"`
+}
+
+func RunInlineQuery2(l *v4.LookerSDK, request RequestRunInlineQuery2, options *rtl.ApiSettings) (string, error) {
+	var result string
+	sdk := reflect.ValueOf(l).Elem()
+	session := sdk.FieldByName("session")
+	session = reflect.NewAt(session.Type(), unsafe.Pointer(session.UnsafeAddr())).Elem()
```

Comment:
@kurtisvg I wanted to call this out to you specifically.

The session attribute of the LookerSDK is currently private. In order to call an undocumented endpoint I need to use "reflect" and "unsafe" to get the private attribute. Ugh ugly.

The next time I can update the Looker Go SDK I will make session public, and this can all go away.

Reply:
Updated SDK. This awful hack is out.